src/voigt.jl

```diff
@@ -43,24 +47,39 @@ julia> tovoigt(Tensor{4,2}(1:16))
  4  16  12  8
  3  15  11  7
  2  14  10  6
+
+julia> tovoigt(SMatrix, Tensor{4,2}(1:16))
+4×4 SMatrix{4, 4, Int64, 16} with indices SOneTo(4)×SOneTo(4):
+ 1  13   9  5
+ 4  16  12  8
+ 3  15  11  7
+ 2  14  10  6
 ```
+
 """
 function tovoigt end
-@inline function tovoigt(A::Tensor{2, dim, T, M}; order=DEFAULT_VOIGT_ORDER[dim]) where {dim, T, M}
-    @inbounds tovoigt!(Vector{T}(undef, M), A; order=order)
+# default to regular Array
+@inline function tovoigt(A::AbstractTensor; kwargs...)
+    return tovoigt(Array, A; kwargs...)
 end
-@inline function tovoigt(A::Tensor{4, dim, T, M}; order=DEFAULT_VOIGT_ORDER[dim]) where {dim, T, M}
-    @inbounds tovoigt!(Matrix{T}(undef, Int(√M), Int(√M)), A; order=order)
+@inline tovoigt(::Type{Array}, A::SecondOrderTensor; kwargs...) = tovoigt(Vector, A; kwargs...)
+@inline tovoigt(::Type{Array}, A::FourthOrderTensor; kwargs...) = tovoigt(Matrix, A; kwargs...)
+
+@inline function tovoigt(::Type{<:Vector}, A::Tensor{2, dim, T, M}; order=nothing) where {dim, T, M}
+    @inbounds _tovoigt!(Vector{T}(undef, M), A, order)
```

Comment:
Potentially Tensor and SymmetricTensor versions can be merged with this pattern
```suggestion
@inline function tovoigt(::Type{<:Vector}, A::Union{Tensor{2, dim, T, M},SymmetricTensor{2, dim, T, M}}; order=nothing, kwargs...) where {dim, T, M}
    @inbounds _tovoigt!(Vector{T}(undef, M), A, order; kwargs...)
```